backend/data-store/db-client/migrations/20231007221102_project.up.sql

```diff
@@ -104,25 +105,39 @@ create table project.rating (
     "rating_id"     int                 generated always as identity primary key,
     "submission_id" int                 not null,
     "assessor_id"   int                 not null,
-    "criterion_id"  int                 not null,
-    "points"        int                 not null,
     "is_draft"      boolean             not null,
     "type"          project.rating_type not null,
-    "justification" text,
 
     constraint rating_submission_fk
         foreign key (submission_id) references project.submission(submission_id),
 
     constraint rating_assessor_fk
-        foreign key (assessor_id) references person.assessor(assessor_id),
-
-    constraint rating_criterion_fk
-        foreign key (criterion_id) references edition.pem_criterion(pem_criterion_id)
+        foreign key (assessor_id) references person.assessor(assessor_id)
 );
 
 create index rating_submission_id_idx
     on project.rating(submission_id);
 
+-- ocena_czastkowa_PEM
+create table project.partial_rating (
+    "partial_rating_id"     int generated always as identity primary key,
+    "rating_id"             int not null,
+    "last_modified_by_id"   int not null,
+    "criterion_id"          int not null,
+    "points"                int not null,
+    "justification"         text not null,
+    "last_modified"         date,
```

Comment:
Perhaps we could do
```
create table editable (
   modified datetime not null
   changed_by int -- references personId
)
```

and then make every table (that needs it) inherit from this table. Afaik this could be good use case for postgresql inheritance since it adds [parent columns to children](https://www.postgresql.org/docs/current/tutorial-inheritance.html)

Tables that would need this include:
* every table that we need to know date of last modification
* every table that we need to know who is the last person to modify this entity
* every table that can be edited by the user (since `modified` would be used to compare versions when detecting races)

What do you think?

Reply:
Anyway i think it should be `datetime` instead of `date` to detect races

Reply:
I'll check the inheritance, because honestly I've never used it in SQL. 

For the second comment I agree, this is the wrong type.

Reply:
I've used inheritance in postgres but I am not fan of it. Do we really need  _modified datetime not null_  and _modifier_ in every table? I agree for project ratings but in other tables it will be useless.

Reply:
And when we want to do it for only one, two or three tables it doesn't still make 
a sense for me. It increases complexity.

Reply:
Well, let's give it a try. If we decide it's making a mess, we can relatively easily revert it, even having data in the DB.